### PR TITLE
Secure Storage: Support extending file size by truncate operation

### PR DESCRIPTION
**Pull request by jamescfkung**
Signed-off-by: James Kung <james.kung@linaro.org>
Tested-by: James Kung <james.kung@linaro.org> (QEMU, MT8173 EVB)

### CODE REVIEW

**Comment by pascal-brand-st-dev** [comment]
Reviewed-by: Pascal Brand <pascal.brand@linaro.org>
    Tested-by: Pascal Brand <pascal.brand@linaro.org> (QEMU platform)

**Comment by pascal-brand-st-dev** [comment]
Anyone else would like to review?

**Comment by jforissier** [comment]
`Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (HiKey)`

Tested with https://github.com/OP-TEE/optee_test/pull/27 commit https://github.com/jamescfkung/optee_test/commit/0554526ff4215f2beab92177ea0d22e555852a9f.

File: core/tee/tee_fs_common.c [hunk 1]
```diff
@@ -1128,73 +1143,124 @@ tee_fs_off_t tee_fs_common_lseek(struct tee_fs_fd *fdp,
...
+	uint8_t *buf = NULL;
 
 	if (!fdp)
-		goto exit;
+		return -1;
 
-	if (fdp->flags & TEE_FS_O_RDONLY)
-		goto exit;
+	if (fdp->flags & TEE_FS_O_RDONLY) {
+		EMSG("Read only");
```
Comment (jbech-linaro) [inline core/tee/tee_fs_common.c]: A bit vague error message, but OK ...

**Comment by jbech-linaro** [comment]
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

File: core/tee/tee_fs_common.c (hunk 1 above)
Comment (jamescfkung) [inline core/tee/tee_fs_common.c]: The error message would looks like:
ERR TEE-CORE:tee_fs_common_ftruncate:1174: Read only

**Pull request merged**